recipes/diligent-core/all/conanfile.py

```diff
@@ -0,0 +1,169 @@
+                self.info.settings.compiler.runtime = "MT/MTd"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def requirements(self):
+        self.requires("opengl/system")
+
+        self.requires("libjpeg/9d")
+        self.requires("libtiff/4.3.0")
+        self.requires("zlib/1.2.11")
+        self.requires("libpng/1.6.37")
+
+        self.requires("spirv-cross/cci.20210930")
+        self.options["spirv-cross"].namespace = self.options.spirv_cross_namespace
+        self.requires("spirv-headers/1.2.198.0")
+        self.requires("spirv-tools/2021.4")
+        self.requires("vulkan-headers/1.2.198")
+        self.requires("volk/1.2.198")
+        self.requires("glslang/11.7.0")
+        if self.settings.compiler == "apple-clang":
+            self.options["glslang"].shared = True
+
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("xorg/system")
+            if not tools.cross_building(self, skip_x64_x86=True):
+                self.requires("xkbcommon/1.3.0")        
+
+    def _diligent_platform(self):
+        if self.settings.os == "Windows":
+            return "PLATFORM_WIN32"
+        elif self.settings.os == "Macos":
+            return "PLATFORM_MACOS"
+        elif self.settings.os == "Linux":
+            return "PLATFORM_LINUX"
+        elif self.settings.os == "Android":
+            return "PLATFORM_ANDROID"
+        elif self.settings.os == "iOS":
+            return "PLATFORM_IOS"
+        elif self.settings.os == "Emscripten":
+            return "PLATFORM_EMSCRIPTEN"
+        elif self.settings.os == "watchOS":
+            return "PLATFORM_TVOS"
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["DILIGENT_BUILD_SAMPLES"] = False
+        self._cmake.definitions["DILIGENT_NO_FORMAT_VALIDATION"] = True
+        self._cmake.definitions["DILIGENT_BUILD_TESTS"] = False
+        self._cmake.definitions["DILIGENT_NO_DXC"] = True
+        self._cmake.definitions["SPIRV_CROSS_NAMESPACE_OVERRIDE"] = self.options.spirv_cross_namespace
+
+        self._cmake.definitions["ENABLE_RTTI"] = True
+        self._cmake.definitions["ENABLE_EXCEPTIONS"] = True
+
+        self._cmake.definitions[self._diligent_platform()] = True
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rename(src=os.path.join(self.package_folder, "include", "source_subfolder"),
+                     dst=os.path.join(self.package_folder, "include", "DiligentCore"))
```

Comment:
there is something wrong if source_subfolder leaks in installation folders.

Moreover, I see that glew is installed. Why is not in requirements instead?

```
-- Installing: /Users/jenkins/w/BuildSingleReference@2/.conan/data/diligent-core/2.5.1/_/_/package/4aa87b05be6fc8d8150f91cb1394ac934efb1629/lib/source_subfolder/Release/libglew-static.a
-- Installing: /Users/jenkins/w/BuildSingleReference@2/.conan/data/diligent-core/2.5.1/_/_/package/4aa87b05be6fc8d8150f91cb1394ac934efb1629/Licenses/ThirdParty/source_subfolder/GLEW-License.txt
```

Reply:
Diligent uses its own glew implementation.

Reply:
Diligent uses source subdirectory name as suffix for installation. I have found a place where it happens, but in the end it was different for libs and headers, and I just gave up, because renaming is easier and more stable about upstream changes

Reply:
https://github.com/DiligentGraphics/DiligentCore/blob/6de211144c6f1866f544f9580b9d5b295efcb3b9/CMakeLists.txt#L412

Reply:
Should be less of a problem in v2 the project is sensitive to the wrapper